cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -470,6 +485,51 @@ open class Node @Inject constructor(private val project: Project) {
         }
     }
 
+
+    private fun createSchemasCmd() = listOfNotNull(
+            Paths.get(System.getProperty("java.home"), "bin", "java").toString(),
+            "-jar",
+            "corda.jar",
+            "run-migration-scripts",
+            if (allowHibernateToManageAppSchema) "--allow-hibernate-to-manage-app-schema" else null)
+
+    private fun runSchemaMigration(){
+        if (!runSchemaMigration) return
+        project.logger.lifecycle("Run database schema migration scripts${if(allowHibernateToManageAppSchema) " - managing CorDapp schemas with hibernate" else ""}")
+        runNodeJob(createSchemasCmd(), "node-schema-cordform.log")
+    }
+
+    val LOGS_DIR_NAME: String = "logs"
+
+    private fun runNodeJob(command: List<String>, logfileName: String) {
+        val logsDir = Files.createDirectories(nodeDir.toPath().resolve(LOGS_DIR_NAME))
+        val nodeRedirectFile = logsDir.resolve(logfileName).toFile()
+        val process = ProcessBuilder(command)
+                .directory(nodeDir)
+                .redirectErrorStream(true)
+                .redirectOutput(nodeRedirectFile)
+                .apply { environment()["CAPSULE_CACHE_DIR"] = "../.cache" }
+                .start()
+        try {
+            if (!process.waitFor(3, TimeUnit.MINUTES)) {
```

Comment:
Should this interval be configurable?

Reply:
Maybe? It isn't in the bootstrapper, and I think 3 minutes is generous for schema creation - but I can make it configurable for good measure.